src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferDictionaryTryGetValueMethodsTests.cs

```diff
@@ -0,0 +1,541 @@
+            string key = ""key"";
+            IDictionary<string, int> data = new Dictionary<string, int>();
+
+            return data.TryGetValue(key, out var value) ? value : 2;";
+
+        #region NoDiagnostic
+
+        private const string DictionaryContainsKeyModifyDictionary = @"
+            string key = ""key"";
+            IDictionary<string, int> data = new Dictionary<string, int>();
+            if (data.ContainsKey(key))
+            {
+                Console.WriteLine(2);
+                data[key] = 2;
+                Console.WriteLine(data[key]);
+                
+                return 2;
+            }
+
+            return 0;";
+
+        private const string DictionaryContainsKeyNonIDictionary = @"
+            string key = ""key"";
+            MyDictionary<string, int> data = new MyDictionary<string, int>();
+            if (data.ContainsKey(key))
+            {
+                Console.WriteLine(2);
+                Console.WriteLine(data[key]);
+                
+                return 2;
+            }
+
+            return 0;";
+
+        private const string DictionaryContainsKeyNotGuardedByContainsKey = @"
+            string key = ""key"";
+            int value = 3;
+            Dictionary<string, int> data = new Dictionary<string, int>();
+            if (data.ContainsValue(value))
+            {
+                Console.WriteLine(data[key]);
+            }
+
+            return 0;";
+
+        #endregion
+
+        #endregion
+
+        #region VB Tests
+
+        private const string VbTemplate = @"
+Imports System
+Imports System.Collections
+Imports System.Collections.Concurrent
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Test
+    Public Class TestClass
+        Public Function TestMethod() As Integer
+            {0}
+        End Function
+
+        Private Class MyDictionary(Of TKey, TValue)
+            Public Function ContainsKey(ByVal key As TKey) As Boolean
+                Return True
+            End Function
+
+            Default Public Property Item(ByVal key As TKey) As TValue
+                Get
+                    Return Nothing
+                End Get
+                Set(ByVal value As TValue)
+                End Set
+            End Property
+        End Class
+    End Class
+End Namespace";
+
+        private const string VbDictionaryContainsKeyPrintValue = @"
+            Dim key As String = ""key""
+            Dim data As Dictionary(Of String, Integer) = New Dictionary(Of String, Integer)()
+
+            If {|#0:data.ContainsKey(key)|} Then
+                Console.WriteLine({|#1:data(key)|})
+            End If
+
+            Return 0";
+
+        private const string VbDictionaryContainsKeyPrintValueFixed = @"
+            Dim key As String = ""key""
+            Dim data As Dictionary(Of String, Integer) = New Dictionary(Of String, Integer)()
+            
+            Dim value = Nothing
```

Comment:
Seems the default value is set with declaration
```suggestion
            Dim value As Integer 
```